File: src/event/impl/plugin/PluginEvent.php
<?php

namespace pocketcloud\cloud\event\impl\plugin;

use pocketcloud\cloud\event\Event;
use pocketcloud\cloud\plugin\CloudPlugin;

abstract class PluginEvent extends Event {

    public function __construct(protected readonly CloudPlugin $plugin) {}

    public function getPlugin(): CloudPlugin {
        return $this->plugin;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/plugin/PluginEvent.java
package de.pocketcloud.cloud.event.impl.plugin;

import de.pocketcloud.cloud.event.Event;
import de.pocketcloud.cloud.plugin.CloudPlugin;
import lombok.Getter;

public abstract class PluginEvent extends Event {

    @Getter
    private final CloudPlugin plugin;

    public PluginEvent(CloudPlugin plugin) {
        this.plugin = plugin;
    }
}